gcc/rust/checks/errors/rust-const-checker.cc

```diff
@@ -617,7 +617,10 @@ ConstChecker::visit (Function &function)
 {
   auto const_fn = function.get_qualifiers ().is_const ();
   if (const_fn)
-    const_context.enter (function.get_mappings ().get_hirid ());
+    {
+      const_context.enter (function.get_mappings ().get_hirid ());
+      this->in_const_context = true;
```

Comment:
So yeah, no need for that line :) also, you were never setting it back to false, so anytime we'd see one constant function we'd be in a constant context for the rest of the crate we're checking